doc/design/sql_program_dependency_analyze.md

```diff
@@ -0,0 +1,130 @@
+# SQL Program Dependency Analyze
+
+Consider the SQL program shown below, which may be a common case to construct a full
+AI pipeline using pure SQL (with SQLFlow). The below SQL program is trying to:
+
+1. Preprocess the original data and put the data in one temp table for training
+2. Train one DNN model and one XGBoost model.
+3. Run model explanation on both models for later analysis work.
+4. Clean up the training table.
+
+```sql
+-- random sampling 1,000,000 samples from one month's data,
+-- fill in empty data, and put the result to "train_table"
+CREATE TABLE train_table AS
+SELECT COALESCE(field1, 0), COALESCE(NULLIF(field2, ""), 0) label
+FROM user_visits
+WHERE dt<'20200313' and dt>'20200213'
+ORDER BY RAND() LIMIT 1000000;
+-- do model training
+SELECT * FROM train_table
+TO TRAIN DNNClassifier
+WITH model.n_classes=2, model.hidden_units=[4096,1024,256]
+LABEL 'label'
+INTO my_dnn_model;
+-- train another model using XGBoost
+SELECT * FROM train_table
```

Comment:
Confusing why training xgb/dnn in one AI job with the same dataset.

Reply:
There was a case that needs to train a tree model and a linear model in one job and analyze the feature importance of both models.